libp2p/Session.cpp

```diff
@@ -305,6 +327,44 @@ void Session::write()
 	});
 }
 
+void Session::writeFrames()
+{
+	bytes const* out = nullptr;
+	DEV_GUARDED(x_writeQueue)
+	{
+		if (m_encframes.empty() || m_encframes[0].empty())
```

Comment:
consider removing empty frames at he beginning - might there be a following item which is non-empty?